evalml/data_checks/class_imbalance_data_check.py

```diff
@@ -33,29 +34,33 @@ def validate(self, X, y):
             y: Target labels to check for imbalanced data.
 
         Returns:
-            list (DataCheckWarning, DataCheckError): list with DataCheckWarnings if imbalance in classes is less than the threshold,
-                                                     and DataCheckErrors if the number of values for each target is below 2 * num_cv_folds.
+            dict: Dictionary with DataCheckWarnings if imbalance in classes is less than the threshold,
+                  and DataCheckErrors if the number of values for each target is below 2 * num_cv_folds.
 
         Example:
             >>> X = pd.DataFrame({})
             >>> y = pd.Series([0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
             >>> target_check = ClassImbalanceDataCheck(threshold=0.10)
-            >>> assert target_check.validate(X, y) == [DataCheckError("The number of instances of these targets is less than 2 * the number of cross folds = 6 instances: [0]", "ClassImbalanceDataCheck"), DataCheckWarning("The following labels fall below 10% of the target: [0]", "ClassImbalanceDataCheck")]
+            >>> assert target_check.validate(X, y) == {DataCheckMessageType.ERROR: [DataCheckError("The number of instances of these targets is less than 2 * the number of cross folds = 6 instances: [0]", "ClassImbalanceDataCheck")],\
+                                                       DataCheckMessageType.WARNING: [DataCheckWarning("The following labels fall below 10% of the target: [0]", "ClassImbalanceDataCheck")]}
         """
+        messages = {
```

Comment:
I think the downside of using `DataCheckMessageType` as the keys is that the user needs to import the enum in order to look at the warnings and results. I think that can be annoying, especially if the user is running automl and trying to access `data_checks_results`:

1. Creating a `DataCheckResults` class.
2. Using string values like "warning" and "error" as keys in the dict you have here.

The benefits of 1 is that we can expose the warnings and errors as instance properties and the user doesn't have to import another class. We can also have an api for checking if there are warnings or errors as opposed to `if self._data_check_results[DataCheckMessageType.ERROR]`. The downside is that it's another class but our docs already show how to get data from the data checks results.

The benefit of 2 is that it'd be a small change but would definitely lead to typos lol.

My vote is for 1 I think it'd be fine to keep it as-is for now too. I saw that you left `to use DataCheckMessageType as key?` in the PR description so wanted to offer my thoughts lol.

Reply:
Ooo I like this suggestion a lot. I agree, the reason why I left the comment about using `DataCheckMessageType` as key was because as I was updating the code, I too felt the inconvenience / frustration of having to import the enum everywhere, but as @dsherry had mentioned, since we already have the enums in place we might as well use them.

That being said, I like the idea of creating a separate `DataCheckResults` class, and having `warnings` and `errors` as attributes 🤔 That way, the user doesn't need to directly type in the keys as strings, and any typos will result in an `AttributeError` instead.